src/region_plane.cpp

```diff
@@ -174,6 +207,24 @@ void RegPlane::shape_update()
   if (ystyle == VARIABLE) yp = yscale * input->variable->compute_equal(yvar);
 
   if (zstyle == VARIABLE) zp = zscale * input->variable->compute_equal(zvar);
+
+  if (nxstyle == VARIABLE) normal[0] = xscale * input->variable->compute_equal(nxvar);
+
+  if (nystyle == VARIABLE) normal[1] = yscale * input->variable->compute_equal(nyvar);
+
+  if (nzstyle == VARIABLE) normal[2] = zscale * input->variable->compute_equal(nzvar);
+
+  if ((nxstyle == CONSTANT) && (nystyle == CONSTANT) && (nzstyle == CONSTANT)) return;
+
+  // enforce unit normal
+
+  double rsq = normal[0] * normal[0] + normal[1] * normal[1] + normal[2] * normal[2];
+  if (rsq == 0.0)
```

Comment:
I think there is a logic problem with this normalization of the norm vector.  If some components are CONSTANT and some are VARIABLE, then the CONSTANT components are rescaled and will thus not remain constant at each timestep.

I think you either need to have a 2nd copy of the norm vector (initial and normalized) so that the initial copy can persist.  Or you need to force the user to make all components constant or all components variable.  Not sure what your intent is.

Reply:
> Or you need to force the user to make all components constant or all components variable

Since this is just *one vector*, this requirement seems very reasonable to me.